Source/Core/VideoCommon/VideoBackendBase.cpp

```diff
@@ -237,6 +238,65 @@ void VideoBackendBase::PopulateBackendInfo()
   g_Config.Refresh();
 }
 
+std::optional<std::string> VideoBackendBase::GetWarningMessage()
+{
+  // Software renderer warning.
+  if (g_video_backend->GetName() == "Software Renderer")
+  {
+    return Common::GetStringT("The software renderer is significantly slower than other "
+                              "backends and is only recommended for debugging purposes.");
+  }
+
+  // Feature checks.
+  std::vector<std::string> missing_features;
+  if (!g_Config.backend_info.bSupportsLogicOp)
+    missing_features.push_back(Common::GetStringT("Framebuffer Logical Operations"));
```

Comment:
Just as a note, these strings are most likely going to be very hard to translate in a good way, which both makes things annoying for translators and potentially could lead to inaccurate translations. Leaving them untranslated might not hurt the user experience much since the average user probably won't understand these strings very well anyway.

Reply:
Fair point. I wasn't sure whether to translate them in the first place, but that makes sense.